jobs/gorouter/spec

```diff
@@ -282,6 +282,18 @@ properties:
     example:
     - name: X-Vcap-Request-Id
     - name: Accept-Ranges
+  router.hop_by_hop_headers_to_filter:
+    description: |
+       (optional, array value) List of HTTP Headers that are filtered for Hop-By-Hop Connection header
```

Comment:
I think we should expand this description. I'm not quite sure what this means.

Given your comments on the other PR:
> I am happy to change the name from bannedList to anything folks think is more easily understood, but essentially those header values would be banned from Client side on Connection header and are subject to removal. I know for me personally, headersToKeep would make it sound like a property that would need to have every single header used in gorouter.

From: https://github.com/cloudfoundry/gorouter/pull/356#discussion_r1286054653

> I think between the two options, I would prefer headersToKeep probably because the values in this field are to be removed from Connection header instead of kept.

From: https://github.com/cloudfoundry/gorouter/pull/356#discussion_r1287040572


Maybe we could add a description like:
> (optional, array value) List of HTTP Headers that are filtered for Hop-By-Hop Connection header.
When clients make requests to the gorouter, these HTTP Headers will be automatically be removed their request's `Connection` header if present before gorouter proxies the request.

Reply:
Updated description